### PR TITLE
Add support for getting problems by tag

### PR DESCRIPTION
**Pull request by jkurian**
[Partial fix for #18](https://github.com/dragfire/leetup/issues/18)
* Added CLI `-t` option for tags which can be used as follows `leetup -t <TAG>` eg) `leetup -t array` 
* I just fetch all the question_ids associated to a tag, and then filter all the questions based on these ids
* If no results are found for tag, I just show some text that says there were no questions found for that tag
* I did not validate the input tag but this can also be done by fetching all the tags available and making sure input tag is one of them

### CODE REVIEW

**Comment by dragfire** [comment]
Someone commented on reddit regarding the use of question_id: https://reddit.com/r/rust/comments/jq0ljz/_/gbo3oq8/?context=1
Maybe we want ‘frontend_question_id’?

File: src/service/leetcode.rs [hunk 1]
```diff
@@ -301,11 +301,71 @@ impl<'a> ServiceProvider<'a> for Leetcode<'a> {
...
-    fn list_problems(&mut self, list: List) -> Result<()> {
+    fn list_problems_with_tag(&mut self, list: List) -> Result<()> {
+        let query = r#"
+            query getTopicTag($slug: String!) {
+                 topicTag(slug: $slug) {
+                   name
+                   translatedName
+                   slug
+                   questions {
+                     questionId
```
Comment (dragfire) [inline src/service/leetcode.rs:312]: Probably we want frontendQuestionId?

File: src/service/leetcode.rs [hunk 2]
```diff
@@ -301,11 +301,71 @@ impl<'a> ServiceProvider<'a> for Leetcode<'a> {
...
+
+        let response = client::post(self, &self.config.urls.graphql, &body, || None)?;
+
+        let question_ids: Option<HashSet<_>> = response["data"]["topicTag"]["questions"]
+            .as_array()
+            .and_then(|questions| {
+                questions
+                    .iter()
+                    .map(|serde_string| {
+                        serde_json::from_value(serde_string["questionId"].clone()).unwrap()
```
Comment (dragfire) [inline src/service/leetcode.rs:335]: Same here!

File: src/service/leetcode.rs [hunk 3]
```diff
@@ -301,11 +301,71 @@ impl<'a> ServiceProvider<'a> for Leetcode<'a> {
...
+
+    fn list_problems(&mut self, list: List, options: Option<ListProblemsOptions>) -> Result<()> {
         let problems_res = self.fetch_all_problems()?;
         let mut probs: Vec<StatStatusPair> =
             serde_json::from_value(problems_res["stat_status_pairs"].clone())?;
 
+        if let Some(question_ids) = options.and_then(|options| options.question_ids) {
+            probs = probs
+                .into_iter()
+                .filter(|question| question_ids.contains(&question.stat.question_id.to_string()))
```
Comment (dragfire) [inline src/service/leetcode.rs:365]: Here

File: src/service/leetcode.rs (hunk 1 above)
Comment (jkurian) [reply src/service/leetcode.rs:312]: Hmm, when I try frontendQuestionId I get a 400 error. Any idea if there is another name for this node?

Comment (dragfire) [reply src/service/leetcode.rs:312]: Try `questionFrontendId`

Comment (jkurian) [reply src/service/leetcode.rs]: Worked! Thanks

**Comment by dragfire** [comment]
Looks good. Thanks!

**Pull request merged**